### ISSUE TITLE
Unable to make TTS call with latest version

### ISSUE THREAD

**Issue by sublime392** [Nexmo/nexmo-php#235]
Attempting a TTS call results in error:
Argument 1 passed to Nexmo\Call\Collection::__invoke() must be an instance of Nexmo\Call\Filter or null, instance of Nexmo\Client\Factory\MapFactory given

The error points to these lines in `Nexmo\Client\Factory\MapFactory` (specifically line 75)
```
            if (is_callable($instance)) {
                 $instance = $instance($this);
             }
```

that were added in commit https://github.com/Nexmo/nexmo-php/commit/88e527326111c53f49ae064fdc2545f226222b82#diff-1cc415cd5895fe9ae5304427b1584683

Is there something I need to change in order for this new version to work, or is it broken?

**Comment by dragonmantank** [comment]
Nothing should have changed, but that was part of a larger refactor. Can you share the block of code you are using to kick off the voice call and I'll look at this ASAP?

**Comment by sublime392** [comment]
I am using Laravel, so I am using the wrapper: https://github.com/Nexmo/nexmo-laravel (the actual composer.json entry is `"nexmo/laravel": "^2.0"`).

The code snippet is
```
    Nexmo::calls()->create([
      'to' => [[
        'type' => 'phone',
        'number' => $to
      ]],
      'from' => [
        'type' => 'phone',
        'number' => config('services.nexmo.sms_from')
      ],
      'answer_url' => [$answerUrl],
      'event_url' => [$answerUrl]
    ]);
```
The `$answerUrl` is a short lived url created at runtime, so I don't have an example of that part.

Anything else that you need?

**Comment by dragonmantank** [comment]
Nope, that's all good, will have a fix for this shortly!

**Comment by dragonmantank** [comment]
#236 is in the review pipeline

**Comment by sublime392** [comment]
That was fast!

**Comment by dragonmantank** [comment]
As a side note, the `calls()` client _is_ deprecated now, but your existing code should work 100%, so this is definitely a bug. 

In v3.0.0 (and v2.2.0) there is a new interface for calls under the `voice()` client. The newer code will look like this:

```php
use Nexmo\Voice\Webhook;
use Nexmo\Voice\OutboundCall;
use Nexmo\Voice\Endpoint\Phone;

$call = (new OutboundCall(new Phone($to), new Phone($from)))
    // Second parameter of webhook can be 'GET' or 'POST', defaults to 'POST', so adjust for your endpoints
    ->setAnswerWebhook(new Webhook($answerUrl))
    ->setEventWebhook(new Webhook($answerUrl))
;
Nexmo::voice()->createOutboundCall($call);
```

**Comment by dragonmantank** [comment]
This should now be fixed with [v2.2.2](https://github.com/Nexmo/nexmo-php/releases/tag/2.2.2), feel free to re-open if still having issues.

Thanks!

**Comment by sublime392** [comment]
Such glorious resolution time! Thanks!
